### ISSUE TITLE
[Autodetect] bind controllers when a button is pressed

### ISSUE THREAD

**Issue by fr500** [libretro/RetroArch#2060]
In many ways the android input driver is now the best for autoconfiguration, because it binds controllers based on whether a button was pressed or not, making it easy to use different controllers depending on what you're playing.

That's not the case on other drivers like dinput/xinput, pads are enumerated when initializing the driver and assigned based on internal IDs that are in no way predictable.

It could be great if all the drivers would bind the controllers once a button has been pressed.

**Comment by magwe** [comment]
I was going to suggest this feature as well.

FWIW I am a Linux user.

I would like to be able to navigate RetroArch using any controller (which is currently not working) and launch a game. After the game has been launched I would like to press a button on any input device and the first device to register a button press should be assigned to player 1. The second device to register a button press should be assigned to player 2 etc. Analog inputs should be ignored as these may have jitter.

**Comment by emko** [comment]
yes and also when your in a game and load the retroarch menu there should be a option there to redo the assignments if you need to, this way you can change what controller your using or reorder the player 1 and 2

**Comment by emko** [comment]
how much do we have to pay to get this? i really want to get rid of my keyboard at least we should allow all controllers to access the menu system

**Comment by ttheuer** [comment]
If I may add a request for a sub-feature (just off the top of my head):

Add an option `Reassign Controllers` (or similar) to the *Quick Menu* and/or *Settings > Input*, where you can trigger this process deliberately.  
Give it a popup that says "Player $X: Press a button" with a timeout.  
If it times out on Player 1, change nothing; if it times out at player N or an already assigned player presses their *Cancel* button, assume there are N-1 players.

That'd be super sweet for quickly changing controllers or just swapping players.
